pageserver/src/tenant/timeline.rs

```diff
@@ -1473,16 +1459,44 @@ impl Timeline {
     }
 
     pub(super) fn launch_wal_receiver(
```

Comment:
```suggestion
    /// Creates and starts the wal receiver.
    ///
    /// This function is expected to be called at most once per Timeline's lifecycle
    /// when the timeline is activated.
    fn launch_wal_receiver(
```

Timeline::activate is the only callsite. It'd be nice to even inline this because of the assert, which rides on the at most 1 activation per timeline.

Inlining does seem to require work. Config would have to be on it's own type, with a single getter, and that'd cut down the LOC required enough I think. But yeah, this PR is simple enough, not necessary to complicate it.